### PR TITLE
Adding LLM, SQLite, KeyValue and Variables

### PR DESCRIPTION
**Pull request by ThorstenHans**
This PR adds APIs for the following capabilities:

- LLM inferencing
- Key Value Stores
- SQLite Databases
- Variables

### CODE REVIEW

**Comment by ThorstenHans** [comment]
Added MQTT in additional commit (ab79491)

File: src/KeyValue/KeyValue.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+using System.Text;
+
+namespace Spin.SDK.KeyValue;
+
+public class Store
+{
+    private readonly SpinHttpWorld.wit.imports.fermyon.spin.v2_0_0.IKeyValue.Store _store;
```
Comment (dicej) [inline src/KeyValue/KeyValue.cs]: Would it make sense to add a `using SpinHttpWorld.wit.imports.fermyon.spin.v2_0_0;` to this file so we don't have to use the fully-qualified type names?

File: src/KeyValue/KeyValue.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        var raw = Get(key);
+        return raw == null ? null : encoding.GetString(raw);
+    }
+
+    public void Set(string key, byte[] value)
+    {
+        _store.Set(key, value);
+    }
+
+    public void GetString(string key, string value, Encoding encoding)
```
Comment (dicej) [inline src/KeyValue/KeyValue.cs]:
```suggestion
    public void SetString(string key, string value, Encoding encoding)
```

File: src/LLM/InferencingParams.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+namespace Spin.SDK.LLM;
+
+public class InferencingParams
+{
+    public uint MaxTokens { get; set; }
```
Comment (dicej) [inline src/LLM/InferencingParams.cs]: Am I correct that all these properties will default to zero values (and remain that way unless the programmer remembers to set them)?  If so, should we change the defaults to more useful values?  And/or add a constructor that requires some of them to be set by the programmer?

**Review by dicej — APPROVED** [review]
Looks great overall; just a few suggestions and questions.  Thanks for doing this!

File: src/LLM/InferencingParams.cs (hunk 3 above)
Comment (ThorstenHans) [reply src/LLM/InferencingParams.cs]: that makes sense. I'll check with other SDKs to see which default value we used there

File: src/KeyValue/KeyValue.cs (hunk 1 above)
Comment (ThorstenHans) [reply src/KeyValue/KeyValue.cs]: I changed this to have the following using statement

```csharp
using SpinHttpWorld.wit.imports.fermyon.spin.v2_0_0;
```

This is due to increased readability of higher level `Store` vs generated `IKeyValue.Store`.

File: src/LLM/InferencingParams.cs (hunk 3 above)
Comment (ThorstenHans) [reply src/LLM/InferencingParams.cs]: Adopted default values from JS SDK (see https://github.com/fermyon/spin-js-sdk/blob/e70f94396ee7d5caa6107c84039903f05ba351d3/src/llm.ts#L98 and following)

**Pull request merged**